scripts/delete-unused-users.py

```diff
@@ -100,70 +96,115 @@ def should_delete(user, inactive_since):
+def delete_users_from_hub(hub_url, token, inactive_since, dry_run=False):
+    """Delete users from a provided hub url"""
+    headers = {
+        "Accept": "application/jupyterhub-pagination+json",
+        "Authorization": f"Bearer {token}",
+    }
+
+    print(f"Scanning for users eligible for deletion on hub: {hub_url}")
+    users = list(retrieve_users(hub_url, headers, inactive_since))
+    print(f"Flagged {len(users)} users for deletion on hub: {hub_url}")
+
+    count = 1
+    for user in users:
+        if not dry_run:
+            delete_user(hub_url, headers, user['name'])
+            print(f"{count}: deleting {user['name']}")
+        else:
+            print(f"Skipped {user['name']} due to dry run.")
+        count += 1
+
+    count -= 1
+    if not dry_run:
+        print(f"Deleted {count} total users from the ORM for hub: {hub_url}.")
+    else:
+        print(f"Would have deleted {count} total users from the ORM for hub: {hub_url}.")
+
 def main(args):
     """
     Get users from a hub, check to see if they should be deleted from the ORM
     and if so, delete them!
     """
-    count = 1
-    for user in list(retrieve_users(args.hub_url, args.inactive_since)):
-        print(f"{count}: deleting {user['name']}")
-        count += 1
-        if not args.dry_run:
-            delete_user(args.hub_url, user['name'])
-        else:
-            logger.warning(f"Skipped {user['name']} due to dry run.")
+    if not args.credentials:
+        logger.info("No credentials file, attempting operations on a single hub.")
+        if not args.hub_url:
+            logger.error("When not using the credentials file, you must specify a hub with the --hub_url argument.")
+            raise
+
+        logger.debug(f"Checking for and deleting ORM users on hub: {args.hub_url}")
+        token = os.environ["JUPYTERHUB_API_TOKEN"]
+        delete_users_from_hub(args.hub_url, token, args.inactive_since, args.dry_run)
+
+    elif args.credentials:
+        logger.info(f"Attempting to use credentials file: {args.credentials}")
+        creds = json.loads(open(args.credentials).read())
+        if not creds:
+            logger.error(f"The credentials file is empty: {args.credentials}")
+            raise
+
+        for hub in creds.keys():
+            logger.debug(f"Checking for and deleting ORM users on hub: {hub}")
+            token = creds[hub]
+            delete_users_from_hub(hub, token, args.inactive_since, args.dry_run)
+
+    else:
+        logger.error("You must specify a single hub with the --hub_url argument, or a json file containing hubs and api keys with the -f argument.")
+        raise
 
-    count -= 1
-    print(f"Deleted {count} total users from the ORM.")
 
 if __name__ == "__main__":
     argparser = argparse.ArgumentParser()
+    argparser.add_argument(
+        '-f',
+        dest='credentials',
+        help='Path to a json file containing hub url and api keys.  Format is: {"hub1_url": "hub1_key", "hub2_url":, "hub2_key"}'
```

Comment:
The json format appears to use data as a key, which is concise. However I'd recommend that the keys not be data, e.g. `[{"url": "https://...", "key": "..."}, {"url": "https://...", "key": "..."}]`. Not strictly necessary of course, but good practice.

Reply:
this way, the config is much more concise and looks cleaner/easier to read:

```
{
  "https://datahub.berkeley.edu": "XXXXXXXX",
  "https://logodev.datahub.berkeley.edu": "XXXXXXXX",
  "https://a11y.datahub.berkeley.edu": "XXXXXXXX",
  ...
}
```

Reply:
Fair enough. For this little app it is fine, but you couldn't write a schema for this for example.

Reply:
noted!